libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackAdapter.cs

```diff
@@ -427,18 +444,26 @@ public async Task ProcessAsync(HttpRequest request, HttpResponse response, IBot
                                 {
                                     Id = null,
                                 },
-                                ChannelData = slackEvent,
+                                ChannelData = new NewSlackMessage()
```

Comment:
Same here about repeating code (check lines `372` to `377`)

Reply:
Maybe you could have an abstraction to handle this, it is hard to mantain otherwise.